### PR TITLE
added checks for GRPC because it expects ASCIIString

### PR DESCRIPTION
**Pull request by dhilpipre**
### Overview
When the Java Agent attempts to read headers from io.grpc.netty.GrpcHttp2HeadersUtils.GrpcHttp2RequestHeaders it throws an exception because the get method expects an io.netty.util.AsciiString input but receives a String from the agent.  

### Related Github Issue
This fixes
https://github.com/newrelic/newrelic-java-agent/issues/2177

This code check to see if the Http2Headers class that is passed into the constructor for com.agent.instrumentation.netty4116.Http2RequestHeaderWrapper is GrpcHttp2RequestHeaders.  If it is it sets a flag.   
In both the getHeader and getHeaders methods, it checks if the GRPC flag is set and if true then creates an AsciiString to use.

### CODE REVIEW

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java [hunk 1]
```diff
@@ -28,12 +29,14 @@
 
 public class Http2RequestHeaderWrapper extends ExtendedRequest {
     private static final Pattern URL_REPLACEMENT_PATTERN = Pattern.compile("(?i)%(?![\\da-f]{2})");
+    private static final String GRPCHEADERSUTIL = "io.grpc.netty.GrpcHttp2HeadersUtils.GrpcHttp2RequestHeaders";
```
Comment (jtduffy) [inline instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java:32]: Can this be changed to `GRPC_HEADER_UTIL_CLASS`?

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java [hunk 2]
```diff
@@ -124,6 +128,13 @@ public String getRequestURI() {
     @Override
     public String getHeader(String name) {
         try {
+        	
+        	if(isGRPCHttp2) {
```
Comment (jtduffy) [inline instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java:132]: Indent if contents with 4 spaces

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java [hunk 3]
```diff
@@ -134,7 +145,10 @@ public String getHeader(String name) {
                 return http2Headers.get(lowerCaseHeaderName).toString();
             }
         } catch (Exception e) {
-            AgentBridge.getAgent().getLogger().log(Level.FINER, e, "Unable to get Http2Headers header: {0}", e.getMessage());
+        	String errorMsg = e.getMessage();
```
Comment (jtduffy) [inline instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java:148]: Indent with 4 spaces

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java [hunk 4]
```diff
@@ -205,12 +219,22 @@ public String getHost() {
     public List<String> getHeaders(String name) {
         List<String> headers = new ArrayList<>();
         try {
-            // HTTP/2 only supports lowercase headers
-            String lowerCaseHeaderName = name.toLowerCase();
-            List<CharSequence> allHeaders = http2Headers.getAll(lowerCaseHeaderName);
-            for (CharSequence header : allHeaders) {
-                headers.add(header.toString());
-            }
+            if (!isGRPCHttp2) {
```
Comment (jtduffy) [inline instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java:222]: Indention

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java [hunk 5]
```diff
@@ -205,12 +219,22 @@ public String getHost() {
...
+				String lowerCaseHeaderName = name.toLowerCase();
+				List<CharSequence> allHeaders = http2Headers.getAll(lowerCaseHeaderName);
+				for (CharSequence header : allHeaders) {
+					headers.add(header.toString());
+				} 
+			} else {
+				// GRPC only accepts AsciiString
+				AsciiString asciiName = new AsciiString(name);
+				List<CharSequence> allHeaders = http2Headers.getAll(asciiName);
+				for (CharSequence header : allHeaders) {
```
Comment (jtduffy) [inline instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java:233]: Can the `for()` loop be re-factored out of the if/else` blocks?

**Review by jtduffy — CHANGES_REQUESTED** [review]

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java (hunk 5 above)
Comment (dhilpipre) [reply instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java]: done

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java (hunk 3 above)
Comment (dhilpipre) [reply instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java]: all tabs replaced with 4 spaces

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java (hunk 2 above)
Comment (dhilpipre) [reply instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java]: all tabs replaced with 4 spaces

File: instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java (hunk 1 above)
Comment (dhilpipre) [reply instrumentation/netty-4.1.16/src/main/java/com/agent/instrumentation/netty4116/Http2RequestHeaderWrapper.java]: done

**Review by jtduffy — APPROVED** [review]

**Pull request merged**